libs/sdk-ui-pivot/src/impl/structure/colLocatorMatching.ts

```diff
@@ -0,0 +1,101 @@
+             *
+             * Once code has that, it can compare the URI of attribute element for which this group was created with
+             * the excepted attribute element in the locator.
+             */
+            const groupByAttributeLocalId = col.attributeDescriptor.attributeHeader.localIdentifier;
+            const matchingLocator = locators.find((locator) => {
+                return (
+                    isAttributeColumnLocator(locator) &&
+                    locator.attributeLocatorItem.attributeIdentifier === groupByAttributeLocalId
+                );
+            });
+
+            if (!matchingLocator) {
+                // if there is no matching attribute locator yet code is on data col group, then it
+                // means there are less attributes in the table than there are attribute locators. the
+                // table has changed yet some sort items hang around. bail out immediately with no match.
+                return undefined;
+            }
+
+            const elementToMatch = (matchingLocator as IAttributeColumnLocator).attributeLocatorItem.element;
+            // while the attribute locator has element optional (for wildcard match), all the remaining code always populates
+            // the attribute element.
+            // TODO: revise the API, it may be that it really should not be optional in the interface.
+            invariant(elementToMatch);
+
+            if (col.header.attributeHeaderItem.uri === elementToMatch) {
+                if (locators.length === 1) {
+                    // elements match; see if all locators exhausted. if so, it means the width item does
+                    // not contain any measure locator; it's OK to match the DataColGroup though, so mark it
+                    // as found and bail out
+
+                    found = col;
+                } else {
+                    // otherwise dive deeper to inspect the children. make sure not to send the already-matched locator.
+                    const remainingLocators: ColumnLocator[] = locators.filter(
+                        (locator) => locator !== matchingLocator,
+                    );
+
+                    found = searchForLocatorMatch(col.children, remainingLocators);
+                }
+            }
+        } else if (isDataColLeaf(col)) {
+            if (locators.length > 1 || !isMeasureColumnLocator(locators[0])) {
+                // code has reached the leaves. at this point there must be just one locator left and it must
+                // be the measure column locator. if this is not the case, then there are too many locators -
+                // the table has likely changed and old locators were left around. bail out immediately.
+
+                return undefined;
+            }
+
+            if (colMeasureLocalId(col) === locators[0].measureLocatorItem.measureIdentifier) {
+                found = col;
+            }
+        }
+
+        if (found) {
```

Comment:
Can we return the `found` items immediately? Or is there a way the `found` is re-assigned during one pass of the loop (I don't see one at first glance)?